src/Microsoft.Health.Fhir.SqlServer/Features/Storage/SqlServerSchemaMigrationDataStore.cs

```diff
@@ -0,0 +1,51 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System.Data.SqlClient;
+using System.Threading;
+using System.Threading.Tasks;
+using EnsureThat;
+using Microsoft.Extensions.Logging;
+using Microsoft.Health.Fhir.SqlServer.Features.Exceptions;
+using Microsoft.Health.Fhir.SqlServer.Features.Schema;
+using Microsoft.Health.Fhir.SqlServer.Features.Schema.Model;
+
+namespace Microsoft.Health.Fhir.SqlServer.Features.Storage
+{
+    internal class SqlServerSchemaMigrationDataStore : ISchemaMigrationDataStore
+    {
+        private readonly SqlConnectionWrapperFactory _sqlConnectionWrapperFactory;
+        private readonly ILogger<SqlServerSchemaMigrationDataStore> _logger;
+
+        public SqlServerSchemaMigrationDataStore(
+            SqlConnectionWrapperFactory sqlConnectionWrapperFactory,
+            ILogger<SqlServerSchemaMigrationDataStore> logger)
+        {
+            EnsureArg.IsNotNull(sqlConnectionWrapperFactory, nameof(sqlConnectionWrapperFactory));
+            EnsureArg.IsNotNull(logger, nameof(logger));
+
+            _sqlConnectionWrapperFactory = sqlConnectionWrapperFactory;
+            _logger = logger;
+        }
+
+        public async Task<int> GetLatestCompatibleVersionAsync(CancellationToken cancellationToken)
+        {
+            using (SqlConnectionWrapper sqlConnectionWrapper = _sqlConnectionWrapperFactory.ObtainSqlConnectionWrapper(true))
```

Comment:
Does this actually need to enlist in a transaction? Is there an existing transaction?